# HG changeset patch
# User Olli Saarela <olli.saarela@gmail.com>
# Date 1249504510 14400
#      Wed Aug 05 16:35:10 2009 -0400
# Node ID ac5e381f7a1c5060a097b1f372aa72cfa19e6db9
# Parent  3aeb7d881578cd1aa4ad19d6001f6036d2575467
__gnuplot_get_var__: if read fails to return data, sleep before trying again

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2009-08-05  Olli Saarela  <olli.saarela@gmail.com>
+
+	* plot/__gnuplot_get_var__.m: If read fails to return data, sleep
+	before trying again.
+
 2009-08-04  Pieter Eendebak <pieter.eendebak@gmail.com> 
 
 	* set/setxor.m: Support cell arrays of strings.
 
 2009-08-03  Jaroslav Hajek  <highegg@gmail.com>
 
 	* general/sub2ind.m: Remove source.
 	* general/ind2sub.m: Remove source.
diff --git a/scripts/plot/__gnuplot_get_var__.m b/scripts/plot/__gnuplot_get_var__.m
--- a/scripts/plot/__gnuplot_get_var__.m
+++ b/scripts/plot/__gnuplot_get_var__.m
@@ -122,17 +122,19 @@ function gp_var_value = __gnuplot_get_va
       fflush (ostream);
       ## Direct gnuplot to print to <STDERR>
       fputs (ostream, "set print;\n");
       fflush (ostream);
 
       str = {};
       while (isempty (str))
         str = char (fread (istream)');
-        if (! isempty (str))
+        if (isempty (str))
+	  sleep (0.05);
+	else
           str = regexp (str, "OCTAVE:.*", "match");
           str = str{end}(8:end);
         endif
         fclear (istream);
       endwhile
     endif
 
     ## Strip out EOLs and the continuation character "|"
